File: 30_days_of_coding/Day_12_Inheritance.cs
class Person{
	protected string firstName;
	protected string lastName;
	protected int id;
	
	public Person(){}
	public Person(string firstName, string lastName, int identification){
			this.firstName = firstName;
			this.lastName = lastName;
			this.id = identification;
	}
	public void printPerson(){
		Console.WriteLine("Name: " + lastName + ", " + firstName + "\nID: " + id); 
	}
}

class Student : Person{
    private int[] testScores;  
  
    /*	
    *   Class Constructor
    *   
    *   Parameters: 
    *   firstName - A string denoting the Person's first name.
    *   lastName - A string denoting the Person's last name.
    *   id - An integer denoting the Person's ID number.
    *   scores - An array of integers denoting the Person's test scores.
    */
    // Write your constructor here
    //public Student(string firstName, string lastName, int id, int[] scores){
    //    this.firstName = firstName;
    //    this.lastName = lastName;
    //    this.id = id;
    //    this.testScores = scores;
    //}
    
    public Student(string firstName, string lastName, int id, int[] scores) : base(firstName, lastName, id){
        this.testScores = scores;
    }
	
    /*	
    *   Method Name: Calculate
    *   Return: A character denoting the grade.
    */
    // Write your method here
    public char Calculate(){
        double avrScore = 0;
        for(int i = 0; i < testScores.Length; i++){
               avrScore += testScores[i];
        }
        avrScore /= testScores.Length;
        if(avrScore < 40){
            return 'T';
        }else if(avrScore >= 40 && avrScore < 55){
            return 'D';
        }else if(avrScore >= 55 && avrScore < 70){
            return 'P';
        }else if(avrScore >= 70 && avrScore < 80){
            return 'A';
        }else if(avrScore >= 80 && avrScore < 90){
            return 'E';
        }else{
            return 'O';
        }
    }
}

class Solution {
	static void Main() {
		string[] inputs = Console.ReadLine().Split();
		string firstName = inputs[0];
	  	string lastName = inputs[1];
		int id = Convert.ToInt32(inputs[2]);
		int numScores = Convert.ToInt32(Console.ReadLine());
		inputs = Console.ReadLine().Split();
	  	int[] scores = new int[numScores];
		for(int i = 0; i < numScores; i++){
			scores[i]= Convert.ToInt32(inputs[i]);
		} 
	  	
		Student s = new Student(firstName, lastName, id, scores);
		s.printPerson();
		Console.WriteLine("Grade: " + s.Calculate() + "\n");
	}
}

File: 30DaysOfCodingJAVA/Day_12_Inheritance.java
class Student extends Person{
	private int[] testScores;

    /*	
    *   Class Constructor
    *   
    *   @param firstName - A string denoting the Person's first name.
    *   @param lastName - A string denoting the Person's last name.
    *   @param id - An integer denoting the Person's ID number.
    *   @param scores - An array of integers denoting the Person's test scores.
    */
    // Write your constructor here
    public Student(String firstName, String lastName, int id, int[] scores){
        super(firstName, lastName, id);
        this.testScores = scores;
    }
    /*	
    *   Method Name: calculate
    *   @return A character denoting the grade.
    */
    // Write your method here
    public char calculate(){
        double avrScore = 0;
        for(int i = 0; i < testScores.length; i++){
            avrScore += testScores[i];
        }
        avrScore /= testScores.length;
        if(avrScore < 40){
            return 'T';
        }else if(avrScore >= 40 && avrScore < 55){
            return 'D';
        }else if(avrScore >= 55 && avrScore < 70){
            return 'P';
        }else if(avrScore >= 70 && avrScore < 80){
            return 'A';
        }else if(avrScore >= 80 && avrScore < 90){
            return 'E';
        }else{
            return 'O';
        }
    }
    
}

class Solution {
	public static void main(String[] args) {
		Scanner scan = new Scanner(System.in);
		String firstName = scan.next();
		String lastName = scan.next();
		int id = scan.nextInt();
		int numScores = scan.nextInt();
		int[] testScores = new int[numScores];
		for(int i = 0; i < numScores; i++){
			testScores[i] = scan.nextInt();
		}
		scan.close();
		
		Student s = new Student(firstName, lastName, id, testScores);
		s.printPerson();
		System.out.println("Grade: " + s.calculate() );
	}
}
